tortoise/backends/base/executor.py

```diff
@@ -267,13 +267,15 @@ async def execute_update(
         values = []
         expressions = {}
         for field in update_fields or self.model._meta.fields_db_projection.keys():
-            if not self.model._meta.fields_map[field].pk:
-                instance_field = getattr(instance, field)
-                if isinstance(instance_field, Expression):
+            if not (field_obj := self.model._meta.fields_map[field]).pk:
```

Comment:
Personally I find this syntax hard to read, having `field_obj` outside of the if statement feels better to me but it might be a personal preference.